src/Temporalio/Common/SimplePlugin.cs

```diff
@@ -0,0 +1,167 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Temporalio.Client;
+using Temporalio.Worker;
+
+namespace Temporalio.Common
+{
+    /// <summary>
+    /// A simple plugin that implements both client and worker plugin interfaces.
+    /// </summary>
+    /// <remarks>
+    /// WARNING: This API is experimental and may change in the future.
+    /// </remarks>
+    public class SimplePlugin : ITemporalClientPlugin, ITemporalWorkerPlugin
+    {
+        /// <summary>
+        /// The plugin options.
+        /// </summary>
+        private readonly SimplePluginOptions pluginOptions;
```

Comment:
On most of these types of classes (e.g. client, worker, replayer, etc) we like to make `Options` a public init-only property they can access.